nightguard/ArrayExtension.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  ArrayExtension.swift
+//  nightguard
+//
+//  Created by Florian Preknya on 12/14/18.
+//  Copyright © 2018 private. All rights reserved.
+//
+
+import Foundation
+
+extension Array {
+    
+    /// Returns the first N elements of the sequence that satisfies the given
+    /// predicate.
+    func prefix(_ maxLength: Int, where predicate: (Element) throws -> Bool) rethrows -> Array<Element> {
```

Comment:
Would be nice to have some Units tests here. Another part would be the predictions in my mind. Could be a part where we could get regressions in future releases...

Reply:
Will do.